src/include/function/cast/cast_operations.h

```diff
@@ -57,7 +57,11 @@ struct CastToBlob {
                 common::StringVector::getInMemOverflowBuffer(&resultVector)
                     ->allocateSpace(result.value.len));
         }
-        common::Blob::fromString(input, result.value.getDataWritable());
+        common::Blob::fromString(reinterpret_cast<const char*>(input.getData()), input.len,
+            result.value.getDataWritable());
+        if (!common::ku_string_t::isShortString(result.value.len)) {
```

Comment:
I wonder if we can have a addBlob function in StringVector. If not, merge the check between l55 and l62

Reply:
We can't merge  l62-64 to l55, because we can only copy the string prefix after we have done copying the overflowbuffer.